pkg/referrerstore/oras/authprovider/authproviderfactory.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authprovider
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+var builtInAuthProviders = make(map[string]AuthProviderFactory)
+
+// AuthProviderFactory is an interface that defines methods to create an AuthProvider
+type AuthProviderFactory interface {
+	Create(authProviderConfig AuthProviderConfig) (AuthProvider, error)
+}
+
+// Register adds the factory to the built in providers map
+func Register(name string, factory AuthProviderFactory) {
+	if factory == nil {
+		panic("auth provider factory cannot be nil")
+	}
+	_, registered := builtInAuthProviders[name]
+	if registered {
+		panic(fmt.Sprintf("auth provider factory named %s already registered", name))
+	}
+
+	builtInAuthProviders[name] = factory
+}
+
+// CreateAuthProvidersFromConfig creates the AuthProvider from the provided configuration.
+// If the AuthProviderConfig isn't specified, use the default auth provider
+func CreateAuthProviderFromConfig(authProviderConfig AuthProviderConfig) (AuthProvider, error) {
+	// if auth provider not specified in config, return default provider
+	if authProviderConfig == nil {
+		return builtInAuthProviders[DefaultAuthProviderName].Create(authProviderConfig)
```

Comment:
Is the default auth provider imported in the main.go?

Reply:
It's currently being imported in oras.go which is how it's being initialized every time. Should I add a blank import in main.go?

Reply:
If it is brought transitively, then it should be fine